File: Problems 0000-0099/71. Simplify Path/solution.java
class Solution {
    public String simplifyPath(String path) {
        String[] parts = path.split("/");

        Stack<String> stack = new Stack();
        for (String name : parts) {
            if (name.isEmpty() || name.equals(".")) continue;
            if (name.equals("..")) {
                if (!stack.isEmpty()) stack.pop();
            } else {
                stack.push(name);
            }
        }

        if (stack.isEmpty()) return "/";

        StringBuilder sb = new StringBuilder();

        for (String name : stack) {
            sb.append("/");
            sb.append(name);
        }
        return sb.toString();
    }
}

File: Problems 0000-0099/71. Simplify Path/solution.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = []
        dirs = [name for name in path.split("/") if name]
        
        for name in dirs:
            if name == "..":
                if stack:
                    stack.pop()
            elif name != ".":
                stack.append(name)

        return "/" + "/".join(stack)